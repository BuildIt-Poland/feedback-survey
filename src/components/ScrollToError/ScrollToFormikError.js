import { useEffect } from 'react';

const scrollToRef = (ref, offset = 0) => {
  window.scrollTo({
    top: ref.current.offsetTop - offset,
    left: 0,
    behavior: 'smooth'
  });
};

const ScrollToFormikError = ({ formik, scrollFieldRef }) => {
  const effect = () => {
    if (formik.submitCount > 0 && !formik.isSubmitting && !formik.isValid) {
      scrollToRef(scrollFieldRef, 40);
    }
  };

  useEffect(effect, [formik.submitCount, formik.isSubmitting]);
  return null;
};

export default ScrollToFormikError;
