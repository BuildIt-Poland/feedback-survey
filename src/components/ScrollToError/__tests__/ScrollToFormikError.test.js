import React from 'react';
import { create } from 'react-test-renderer';
import { render, cleanup } from 'react-testing-library';

import ScrollToFormikError from '../ScrollToFormikError';

const props = {
  formik: {
    submitCount: 1,
    isSubmitting: false,
    isValid: false
  },
  scrollFieldRef: {
    current: {
      offsetTop: 100
    }
  }
};

afterEach(cleanup);

describe('COMPONENT - ScrollToFormikError', () => {
  beforeEach(() => {
    global.scrollTo = jest.fn();
  });

  it('renders correctly', () => {
    const component = create(<ScrollToFormikError {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls scrollToRef with correct values', () => {
    render(<ScrollToFormikError {...props} />);

    expect(global.scrollTo).toHaveBeenCalled();
  });
});
