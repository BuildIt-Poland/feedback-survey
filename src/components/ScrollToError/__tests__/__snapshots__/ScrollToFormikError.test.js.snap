// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - ScrollToFormikError renders correctly 1`] = `null`;
